core/include/drivers/stm32_i2c.h

```diff
@@ -0,0 +1,228 @@
+/* SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause) */
+/*
+ * Copyright (c) 2017-2019, STMicroelectronics
+ */
+
+#ifndef __STM32_I2C_H
+#define __STM32_I2C_H
+
+#include <mm/core_memprot.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <util.h>
+#include <types_ext.h>
+
+enum i2c_speed_e {
+	I2C_SPEED_STANDARD,	/* 100 kHz */
+	I2C_SPEED_FAST,		/* 400 kHz */
+	I2C_SPEED_FAST_PLUS,	/* 1 MHz   */
+};
+
+#define STANDARD_RATE				100000
+#define FAST_RATE				400000
+#define FAST_PLUS_RATE				1000000
```

Comment:
These names sound a bit too generic to me, and it looks like they could be moved them to `stm32_i2c.c`, no?

Reply:
You're right. I'll move the defines.